build.yaml

```diff
@@ -2336,3 +2336,38 @@ steps:
     - deploy_batch
     - deploy_shuffler
     - deploy_address
+ - kind: runImage
+   name: test_batch_invariants
+   image:
+     valueFrom: batch_image.image
+   script: |
+     python3 -m pytest --log-date-format="%Y-%m-%dT%H:%M:%S" --log-format="%(asctime)s %(levelname)s %(name)s %(filename)s:%(lineno)d:%(funcName)s %(message)s" --log-cli-level=INFO -s -vv --instafail -k "test_invariants" /io/test/
+   inputs:
+    - from: /repo/batch/test
+      to: /io/
+   timeout: 300
+   secrets:
+    - name: gce-deploy-config
+      namespace:
+        valueFrom: default_ns.name
+      mountPath: /deploy-config
+    - name: test-dev-tokens
+      namespace:
+        valueFrom: batch_pods_ns.name
+      mountPath: /user-tokens
+    - name: ssl-config-batch-tests
+      namespace:
+        valueFrom: default_ns.name
+      mountPath: /ssl-config
+   scopes:
+    - test
+    - dev
+   dependsOn:
+    - create_deploy_config
+    - create_accounts
+    - default_ns
+    - batch_pods_ns
+    - copy_files
+    - batch_image
+    - deploy_batch
+    - test_batch
```

Comment:
Probably should depend on test_hailtop_batch and the batch docs tests. We should update `delete_batch_instances` to depend on this job since this is the new sink of batch tests.

Reply:
I disagree. I want to be able to run this with dev deploy without spinning up all the infrastructure for ci and running test_hailtop_batch. The batch docs tests run everything locally I believe. The hailtop batch tests do run things on the server.

Reply:
I'll make two jobs. One is runIfRequested only that depends on test_batch. The other is the one for test and deploy.

Reply:
ok. I decided you're right and if I am dev deploying, then I'll comment out dependencies by hand.

Reply:
I guess I don't really understand why delete_batch_instances depends on the tests. I anticipated it to delete the instances *before* deploying batch. I tried using it like: `--steps delete_batch_instances,test_batch_...` and was surprised that the instances weren't deleted first.

Reply:
No. The point of delete_batch_instances is to delete instances that are hanging around after the tests have been run once the driver is killed. For example, a worker in an infinite loop.